src/db/spanner/models.rs

```diff
@@ -469,6 +462,65 @@ impl SpannerDb {
         }
     }
 
+    fn map_collection_names<T>(&self, by_id: HashMap<i32, T>) -> Result<HashMap<String, T>> {
+        let mut names = self.load_collection_names(by_id.keys())?;
+        by_id
+            .into_iter()
+            .map(|(id, value)| {
+                names
+                    .remove(&id)
+                    .map(|name| (name, value))
+                    .ok_or_else(|| DbError::internal("load_collection_names get"))
+            })
+            .collect()
+    }
+
+    fn load_collection_names<'a>(
+        &self,
+        collection_ids: impl Iterator<Item = &'a i32>,
+    ) -> Result<HashMap<i32, String>> {
+        let mut names = HashMap::new();
+        let mut uncached = Vec::new();
+        for &id in collection_ids {
+            if let Some(name) = self.coll_cache.get_name(id)? {
+                names.insert(id, name);
+            } else {
+                uncached.push(id);
+            }
+        }
+
+        if !uncached.is_empty() {
+            // TODO only select names that are in uncached.
+            let sql = self.sql_request("SELECT collectionid, name FROM collections")?;
+            let spanner = &self.conn;
+            let session = spanner.session.name.as_ref().unwrap();
+            let results = spanner
+                .hub
+                .projects()
+                .instances_databases_sessions_execute_sql(sql, session)
+                .doit();
+            match results {
+                Ok(results) => match results.1.rows {
+                    Some(rows) => {
+                        rows.iter().for_each(|row| {
+                            let id = row[0].parse::<i32>().unwrap();
+                            let name = row[1].clone();
+                            if uncached.contains(&id) {
+                                names.insert(id, name.clone());
+                                self.coll_cache.put(id, name.clone()).unwrap();
+                            }
+                        });
+                    }
+                    None => return Err(DbErrorKind::CollectionNotFound.into()),
+                },
+                // TODO Return the correct error
+                Err(_e) => return Err(DbErrorKind::CollectionNotFound.into()),
```

Comment:
Are there other errors that could be returned here? Should we include a `dbg!` to dump them?